### DISCUSSION TITLE
Setup Issue

### DISCUSSION THREAD

**Discussion by gibsonfriedman** [DarkflameUniverse/DarkflameServer#1676] [Q&A] [answered]
Hi, I'm having an issue running the `./MasterServer.exe -a` command. I've tried it with and without '.exe' and am getting this error:
[11-12-24 01:26:17 MasterServer.cpp:97] Using net version 171023 
[11-12-24 01:26:17 MasterServer.cpp:99] Starting Master server...
[11-12-24 01:26:17 MasterServer.cpp:100] Version: 2.3.0
[11-12-24 01:26:17 MasterServer.cpp:101] Compiled on: Wed Dec 11 00:18:05 2024
**[11-12-24 01:26:17 MasterServer.cpp:107] Got an error while connecting to the database: Error parsing url: url parsing error : '//' is not present in the url**
[11-12-24 01:26:17 MasterServer.cpp:108] Migrations not run
[11-12-24 01:26:17 MasterServer.cpp:816] Recieved Signal -1

Does anyone know what this could be? I'd appreciate any help I could get :)

**Comment by EmosewaMC** [comment]
it sounds like your sharedconfig.ini does not have the database details filled in.  Please make sure you have information in all of the mysql fields.

**Reply by gibsonfriedman** [reply]
I added 

mysql_host=localhost
mysql_database=darkflame
mysql_username=darkflame
mysql_password=MY-PASSWORD
external_ip=localhost

And ..\..\client to the client_location variables in the sharedconfig file. Is there anything that I might've missed that's causing that?

**Reply by EmosewaMC** [reply]
what is the full path to masterserver.exe from the root repo folder?
(i.e. serverRoot/build/masterserver.exe)

**Reply by gibsonfriedman** [reply]
DarkflameServer/build/MasterServer.exe

**Reply by EmosewaMC** [reply]
do you happen to also have these fields in the other 4 ini files? (master, chat, auth, world)config.ini?
these will take precedence if they exist in those files, even if the values are empty

**Reply by gibsonfriedman** [reply]
I just looked through the other 4 ini files and didn't see any of those fields in them.

**Reply by EmosewaMC** [reply]
what is the path of the ini files you are editing?

**Reply by gibsonfriedman** [reply]
Documents\DarkflameServer\build\config.ini

**Reply by EmosewaMC** [reply]
how are you building the server?  `build.sh`, `cmake ..`, `cmake .. --preset=<example>`?

**Reply by gibsonfriedman** [reply]
I'm just running, "./build.sh"

**Reply by EmosewaMC** [reply]
Is there a folder called `msvc` in the build folder?  If so, can you copy the ini files in build to the msvc folder?

**Reply by gibsonfriedman** [reply]
There's no msvc folder, should I make one and put the ini files in it?

**Reply by EmosewaMC** [reply]
no that shouldn't be necessary, can you get me the git hash you're currently on? `git rev-parse HEAD`

**Reply by gibsonfriedman** [reply]
1dadeeb36f6a97de4219a041e8774f1e936e9935

**Reply by EmosewaMC** [reply]
Are you on windows10 or 11?

**Reply by EmosewaMC** [reply]
if you're on windows 11 (and maybe 10?) is your documents folder stored in onedrive?

**Reply by gibsonfriedman** [reply]
I'm on Windows 11, I don't think I have onedrive setup though. Is there a way I can double check?

**Reply by EmosewaMC** [reply]
try moving the server source folder to not documents or a windows setup folder (desktop, documents, pictures etc.) maybe try `C:/Users/<you>/Darkflame` ?  if you need to re-build the server, delete the build folder except the ini files.

**Reply by gibsonfriedman** [reply]
I tried moving it to C:/Users/myUser/Darkflame and it's still giving me the same error when I run "./MasterServer.exe -a" or "./MasterServer -a".

**Reply by EmosewaMC** [reply]
im not sure whats wrong then.  I just pulled that commit, ran build.sh, copied over my inis and ran ./MasterServe.exe -a without issues

**Reply by gibsonfriedman** [reply]
Would it be worth deleting the build folder and re-building to see if that has an impact?

**Reply by EmosewaMC** [reply]
before you do that, ill heres what my build folder looked like after a build, with the contents of my inis
![image](https://github.com/user-attachments/assets/d8d7729f-ed77-4bbb-9c91-95e2dd734cdc)
![image](https://github.com/user-attachments/assets/048ee7e2-6e18-4ebb-b9d7-302805687d1e)
![image](https://github.com/user-attachments/assets/7323ab29-3480-4c92-995b-53cbfcc88864)

Do you notice any differences in your build directory/ini file?  the one pictured is the one in build with no msvc folder present and no other inis to be found (my windows setup does not use an environment variable).

**Reply by EmosewaMC** [reply]
sorted by file type, so you can compare it easily

**Reply by EmosewaMC** [reply]
do note that roughly next week, we will be making the new default database sqlite for singleplayer and small servers.  This should bypass this issue when thats done and on top of that you wont need to build from source anymore.

**Reply by gibsonfriedman** [reply]
Yeah mine looks the same :/ not sure wherethis error is even coming from.

**Reply by gibsonfriedman** [reply]
> do note that roughly next week, we will be making the new default database sqlite for singleplayer and small servers. This should bypass this issue when thats done and on top of that you wont need to build from source anymore.

Oh gotcha, maybe I'll just wait till that happens and try again :)

**Reply by EmosewaMC** [reply]
do try a re-build but im not sure itll fix the issue.  if you know how to edit C++ source, you could probably toss a log line in MySQLDatabase::Connect and see what url is being used for the hostname.

**Reply by EmosewaMC** [reply]
we are working on the release which will contain the sqlite database for backend. Should be ready soon though.

**Reply by EmosewaMC** [reply, marked as answer]
[The release is ready](https://github.com/DarkflameUniverse/DarkflameServer/releases/tag/v3.0.0). I forgot some instructions in the setup guide, however you can find the updated instructions [here](https://github.com/DarkflameUniverse/DarkflameServer/blob/7b72dcf57f98fa1b0d062ca9cffde792c3a7513e/README.md#setting-up-a-single-player-server). Please let me know if this works for you and also what you thought of the setup steps/guide. I have tested that downloading the release, making the sharedconfig edit and renaming the boot.cfg allows me to connect to a local server.

**Reply by gibsonfriedman** [reply]
Thank you! I'm away from my computer for the next month but will give it another try when I'm back at it :)
